### PR TITLE
Feature/ARCWELL-431 prevent registration based on server config setting

### PR DESCRIPTION
**Pull request by mberlin-arcweb**
added env var for registration
activated the register flow on the server and admin side

### CODE REVIEW

File: .github/workflows/admin-ci.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ env:
   APP_KEY: NOT_AN_REAL_APP_KEY
   ARCWELL_INSTANCE_ID: arcweb-ci
   ARCWELL_INSTANCE_NAME: CI Arcwell Server"
+  ARCWELL_REGISTER_ENABLED: true
```
Comment (timgetz) [inline .github/workflows/admin-ci.yml]: @mberlin-arcweb I updated this list to be alphabetical order last PR.  Please move this to proper order

File: .github/workflows/server-ci.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ env:
   APP_KEY: NOT_AN_REAL_APP_KEY
   ARCWELL_INSTANCE_ID: arcweb-ci
   ARCWELL_INSTANCE_NAME: CI Arcwell Server"
+  ARCWELL_REGISTER_ENABLED: true
```
Comment (timgetz) [inline .github/workflows/server-ci.yml]: @mberlin-arcweb I updated this list to be alphabetical order last PR.  Please move this to proper order

File: packages/admin/src/app/shared/schemas/user.schema.ts [hunk 3]
```diff
@@ -30,6 +30,16 @@ export const UserUpdateSchema = UserSchema.extend({
   roleId: z.string().uuid().optional(),
 }).strict();
 
+// for registering a new user
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export const UserRegisterSchema: any = z
```
Comment (timgetz) [inline packages/admin/src/app/shared/schemas/user.schema.ts]: Can you try removing the `: any` and the the eslint-disable line and see if it works and you don't get an typescript warning.
We should only need `any` when we have lazy loaded schemas inside the schema you are defining.
I've found if we don't declare `any` IDEs handle typing analysis a little better.  If linting complains anyway by removing then the `any` is totally acceptable in this instance.

**Review by timgetz — CHANGES_REQUESTED** [review]

File: .github/workflows/server-ci.yml (hunk 2 above)
Comment (mberlin-arcweb) [reply .github/workflows/server-ci.yml]: fixed

File: packages/admin/src/app/shared/schemas/user.schema.ts (hunk 3 above)
Comment (mberlin-arcweb) [reply packages/admin/src/app/shared/schemas/user.schema.ts:35]: fixed

File: .github/workflows/admin-ci.yml (hunk 1 above)
Comment (mberlin-arcweb) [reply .github/workflows/admin-ci.yml]: fixed

File: packages/admin/src/app/shared/schemas/user.schema.ts (hunk 3 above)
Comment (mberlin-arcweb) [reply packages/admin/src/app/shared/schemas/user.schema.ts]: removing the any part causes an error on typing due to the form values possibly being undefined
to get around it, i added nullable and optional to the schema

**Review by timgetz — DISMISSED** [review]

**Review by timgetz — APPROVED** [review]

**Pull request merged**